src/cpr_data_access/models.py

```diff
@@ -1025,6 +1026,18 @@ def filter_by_language(self, language: str) -> "Dataset":
         """Return documents whose only language is the given language."""
         return self.filter("languages", [language])
 
+    def sample(self, n: int, random_state: int = 42) -> "Dataset":
+        """Samples n documents from the dataset and returns a Dataset object with only those."""
+        random.seed(random_state)
```

Comment:
can we check that `n <= len(self)` here and log a warning if not? same if `n < 1`, as often sampling functions allow a flexible 'number or proportion' argument

Reply:
also maybe add a test to check for this behaviour

Reply:
Good points, added all 🙌